include/secp256k1_silentpayments.h

```diff
@@ -171,6 +171,28 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_create_s
     const unsigned char *input_hash
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
 
+/** Create Silent Payment label tweak and label.
+ *
+ *  Given a recipient's scan private key b_scan and a label integer m, calculate
+ *  the corresponding label tweak and label:
+ *
+ *  label_tweak = hash(b_scan || m)
+ *  label = label_tweak * G
+ *
+ *  Returns: 1 if label tweak and label creation was successful. 0 if an error occured.
+ *  Args:                  ctx: pointer to a context object
+ *  Out:           label_tweak: pointer to the resulting label tweak
+ *   In:  receiver_scan_seckey: pointer to the receiver's scan private key
+ *                           m: label integer (0 is used for change outputs)
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_create_label_tweak(
+    const secp256k1_context *ctx,
+    secp256k1_pubkey *label,
```

Comment:
in "silentpayments: add label tweak calculation routine" (https://github.com/bitcoin-core/secp256k1/pull/1471/commits/2a00e12e58b6afd2b9ea9b3999dedde91f8ff82f):

I was trying to see if we could completely get rid of `secp256k1_pubkey` as a return value, but here it makes sense since this output is likely to be used in the next commit to create the labeled spend pubkey.

Alternatively, we could combine this and the next function into one that takes `scan_seckey`, `spend_pubkey`, and `m` as arguments and returns `label_tweak`, `label33`, `labeled_spend_pubkey33` as arguments. Not convinced this is strictly better, but curious to hear your thoughts.

Reply:
> Are there use-cases for creating labelled addresses given a labelled pubkey without having access to private key data

None that I can think of, but as I thought about this more I realized:

* Its likely a scanner might want to generate a labels cache when recovering a wallet without needing the `labeled_receiver_pubkey`
* A receiver might want to use a custom label scheme (e.g. hashing a customerId for a static deposit address)

For both of those cases, it seems having the two separate routines is best.